core/arch/arm/kernel/boot.c

```diff
@@ -118,30 +118,35 @@ __weak unsigned long plat_get_aslr_seed(void)
 	return 0;
 }
 
-#if defined(_CFG_CORE_STACK_PROTECTOR)
+#if defined(_CFG_CORE_STACK_PROTECTOR) || defined(CFG_WITH_STACK_CANARIES)
 /* Generate random stack canary value on boot up */
-__weak uintptr_t plat_get_random_stack_canary(void)
+__weak void plat_get_random_stack_canaries(void *buf, size_t ncan, size_t size)
 {
-	uintptr_t canary = 0xbaaaad00;
 	TEE_Result ret = TEE_ERROR_GENERIC;
+	size_t i = 0;
+
+	assert(buf && ncan && size);
 
 	/*
 	 * With virtualization the RNG is not initialized in Nexus core.
 	 * Need to override with platform specific implementation.
 	 */
 	if (IS_ENABLED(CFG_NS_VIRTUALIZATION)) {
 		IMSG("WARNING: Using fixed value for stack canary");
-		return canary;
+		memset(buf, 0xab, ncan * size);
+		goto out;
 	}
 
-	ret = crypto_rng_read(&canary, sizeof(canary));
+	ret = crypto_rng_read(buf, ncan * size);
 	if (ret != TEE_SUCCESS)
 		panic("Failed to generate random stack canary");
 
+out:
 	/* Leave null byte in canary to prevent string base exploit */
-	return canary & ~0xffUL;
+	for (i = 0; i < ncan; i++)
+		*((uint8_t *)buf + size * i) &= ~U(0xff);
```

Comment:
Since you're addressing a byte that should be zeroed, how about just assigning `0`?

Reply:
That's much simpler. Would it be better to use `0x00` to make it more readable?

Reply:
0 is 0, prefer `0`.